app/services/user_service.py

```
@@ -41,6 +42,94 @@ class UserService:
+            else:
+                # PostgreSQL and other databases support FOR UPDATE
+                # We select a single user ID with FOR UPDATE to lock the table
+                statement = select(User.id).limit(1).with_for_update()
+                result = self.session.exec(statement).first()
+                if result is None:
+                    # No users exist
+                    return True
+                # At least one user exists
+                return False
+
+            return count == 0
```

Ruff (0.14.5)

Consider moving this statement to an `else` block

(TRY300)